conans/client/toolchain/cmake.py

```diff
@@ -241,6 +257,37 @@ def _deduce_fpic(self):
             return None
         return fpic
 
+    def _get_architecture(self):
```

Comment:
may we re-use an existing function?

Reply:
Re-used. I copied the contents, because my goal is to decouple from the existing code, to be able to move forward without breaking, and some things can be simplified assuming it will be used only in toolchains.